### PR TITLE
Add plot handler

### PR DESCRIPTION
**Pull request by cigrainger**
Adds a function for adding a handler function to update a `VegaLite` plot. Pretty basic -- tracks iterations in the `:x` and a metric in the `:y`.

Before this is ready to go, we need to figure out a better way of handling multiple epochs when we don't know the length of the input data and don't have a `max_iterations` set. Otherwise the `iteration` resets at each new epoch, which causes janky plots as we get duplicate data points per iteration. We could track absolute number of iterations in `Axon.Loop.State`? Maybe I'm missing another way to do this.

### CODE REVIEW

File: lib/axon/loop.ex [hunk 1]
```diff
@@ -1304,6 +1304,64 @@ defmodule Axon.Loop do
...
+        |> Vl.encode_field(:x, "step", type: :quantitative)
+        |> Vl.encode_field(:y, "loss", type: :quantitative)
+        |> Kino.VegaLite.new()
+        |> Kino.render()
+
+      model
+      |> Axon.Loop.trainer(loss, optim)
+      |> Axon.Loop.plot(plot)
+  """
+  def plot(loop, plot, metric \\ "loss", event \\ :iteration_completed, filter \\ :always) do
```
Comment (josevalim) [inline lib/axon/loop.ex:1326]:
```suggestion
  def plot(loop, plot, opts \\ []) do
```

and set those defaults with `Keyword.validate!`?

File: lib/axon/loop.ex [hunk 2]
```diff
@@ -1304,6 +1304,64 @@ defmodule Axon.Loop do
     )
   end
 
+  @doc """
```
Comment (josevalim) [inline lib/axon/loop.ex:1307]:
```suggestion
  @compile {:no_warn_undefined, Kino.VegaLite}

  @doc """
```

File: lib/axon/loop.ex (hunk 1 above)
Comment (seanmor5) [reply lib/axon/loop.ex:1326]: +1, but I would make `metric` a required argument!

Comment (cigrainger) [reply lib/axon/loop.ex:1328]: So like this?

```suggestion
  def plot(loop, plot, metric, opts \\ []) do
```

Comment (seanmor5) [reply lib/axon/loop.ex:1328]: Yep!

**Review by seanmor5 — APPROVED** [review]

**Comment by cigrainger** [comment]
@seanmor5 any thoughts on the issue re: iterations? After one iteration the y value resets and we end up with this. Which is... not what we want. 
image

**Comment by josevalim** [comment]
Why does the y value reset? Wouldn't Vega-Lite keep the history as we push?

**Comment by cigrainger** [comment]
Let's say you're streaming data in. Like in the mnist example. If you leave the default -1 max_iterations and run more than epoch, then the iteration value resets to 0 for each epoch. So you send 0..1800 the first epoch, but then those same y values again the next epoch with different x values.

**Comment by josevalim** [comment]
Would a "epoch_iterations" that increments all iterations across epochs address it?

We could also have additional layers to pinpoint whenever a new iteration starts.

**Comment by josevalim** [comment]
Maybe we could even emulate it now by doing `(epoch-1)*max_iterations + iterations`?

**Comment by cigrainger** [comment]
That's a long way of saying the iteration value is the one _within the epoch_ but we want to be sending the one _total across epochs_.

**Comment by seanmor5** [comment]
Oh yeah the reason it resets is we want to keep a history per epoch, but not recompile the step function

We could also keep the history in handle meta and use that here

**Comment by cigrainger** [comment]
@seanmor5 wdyt?

**Comment by seanmor5** [comment]
Thank you!

**Comment by jonatanklosko** [comment]
Awesome!! @seanmor5 @josevalim should we have `kino_axon` perhaps, or are we fine with optional deps and having it here?

**Pull request merged**